File: api/src/main/java/net/blueshell/api/dto/EventSignUpDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.Valid;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.dto.user.SimpleUserDTO;
import net.blueshell.api.model.survey.Answer;
import org.jetbrains.annotations.NotNull;

import java.util.List;


@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "EventSignUp")
public class EventSignUpDTO extends BaseDTO {
    private Long id;

    @NotNull
    private Long eventId;

    @Valid
    private List<Answer> answers;

    private GuestDTO guest;

    private Long userId;
}

File: api/src/main/kotlin/net/blueshell/api/event/api/dto/EventSignUpDTO.kt
package net.blueshell.api.event.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.Valid
import net.blueshell.api.event.api.dto.GuestDTO
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import net.blueshell.api.survey.api.dto.AnswerDTO
import net.blueshell.api.user.api.dto.SimpleUserDTO

@Schema(name = "EventSignUp")
@net.blueshell.api.event.api.validation.ValidEventSignUp
@net.blueshell.api.event.api.validation.GuestOrUserRequired
data class EventSignUpDTO(
    var eventId: Long? = null,

    @field:Valid
    var answers: MutableList<AnswerDTO> = mutableListOf(),
    var guest: GuestDTO? = null,
    var user: SimpleUserDTO? = null,
    var userId: Long? = null
) : AuditedAutoIdDTO()
